src/main/java/org/jenkinsci/main/modules/sshd/SshCommandFactory.java

```diff
@@ -3,12 +3,11 @@
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
 import hudson.util.QuotedStringTokenizer;
-import jenkins.model.Jenkins;
-import org.apache.sshd.server.Command;
 
 import java.util.AbstractList;
 import java.util.Arrays;
 import java.util.List;
+import org.apache.sshd.server.command.Command;
```

Comment:
So this is not going to be backward-compatible for plugins implementing this extension point, right? How many are affected, and how do you plan to release fixes?

Reply:
The impact is similar to the change to convert it to a module, so the same I will open PRs for all the active maintained plugins.

Reply:
This is the list of plugins affected by the change of package of the class Command

Active plugins

https://github.com/jenkinsci/git-server-plugin
https://github.com/jenkinsci/gerrit-trigger-plugin
https://github.com/jenkinsci/ssh-agent-plugin

outdated plugins

https://github.com/jenkinsci/remote-terminal-access-plugin - 7 years
https://github.com/jenkinsci/ssh-cli-plugin - 9 years

Reply:
yep, I have to make a PR to change to use the new Class, it is only a change of package so it is pretty straightforward. The problem that I have here is that for some reason the sshd-plugin stop generating incremental in Feb, so I can not make those PRs without point then to an incremental.

Reply:
> No applicable permissions for jenkinsci/sshd-plugin, check jenkins-infra/repository-permissions-updater has the right configuration

Reply:
Fixed in https://github.com/jenkins-infra/repository-permissions-updater/pull/1922

Reply:
cool, this weekend I will make all the rest of the changes.

Reply:
nice, incremental is generated

Reply:
PRs to update to Apache Mina 2.6.x

* https://github.com/jenkinsci/git-server-plugin/pull/24
* https://github.com/jenkinsci/ssh-agent-plugin/pull/54
* The Gerrit trigger plugin needs more work (I have started but not finished yet)
  * Configure incrementals
  * Configure parent plugin 4
  * Review all dependency conflicts

Reply:
I've been trying to make time to update the gerrit trigger to latest pom etc. But every time it ends up taking more time than I thought and I need to put it on ice yet again for other priorities :cry: So if you get a pom bump working I will be more than happy to merge and release as soon as I am able :)

Reply:
yep, it is in progress, dealing with the transitive dependencies and the enforcer plugin is the hard part

Reply:
For me it was all the mocked methods that came after the enforcer fixes. Or maybe I was trying to take too big of a bite with a core bump at the same time :)